src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,40 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+
+    this.hidden = hidden;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (Object.getPrototypeOf(animal) !== Carnivore.prototype
+    && animal.hidden !== true) {
+      animal.health -= 50;
+
+      if (animal.health === 0) {
+        const animalIndex = Animal.alive.indexOf(animal);
+
+        Animal.alive.splice(animalIndex, 1);
+      }
+    }
+  }
 }
 
+Animal.alive = [];
```

Comment:
Will be better to move this immediately after the declaration of the Animal class.